### PR TITLE
Add developer QoL changes

### PR DESCRIPTION
**Pull request by michplunkett**
## Describe your changes
Modified the `README`,  added commands for `eslint` and `prettier` (https://github.com/utdata/icj-project-rig/issues/37#issue-539983163), formatted the codebase, added a Pull Request template, added a GitHub action that makes sure all code is formatted/linted before it can be merged, and took care of https://github.com/utdata/icj-project-rig/issues/87.

## Checklist before requesting a review
- [x] All features within the repository still work.

```
michaelp@MacBook-Air-5 icj-project-rig % gulp
[14:23:33] Using gulpfile ~/Documents/GitHub/icj-project-rig/gulpfile.js
[14:23:33] Starting 'default'...
[14:23:33] Starting 'clean'...
[14:23:33] Finished 'clean' after 690 μs
[14:23:33] Starting 'styles'...
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
[14:23:35] Finished 'styles' after 2.07 s
[14:23:35] Starting 'copy'...
[14:23:35] Finished 'copy' after 4.22 ms
[14:23:35] Starting 'lint'...
[14:23:35] Starting 'scripts'...
[14:23:35] Starting 'images'...
[14:23:35] Finished 'lint' after 25 ms
[14:23:36] Finished 'scripts' after 195 ms
[14:23:36] gulp-imagemin: Minified 0 images
[14:23:36] Finished 'images' after 199 ms
[14:23:36] Starting 'nunjucks'...
[14:23:36] Finished 'nunjucks' after 1.6 ms
[14:23:36] Starting 'bake'...
[14:23:36] Finished 'bake' after 2 ms
[14:23:36] Finished 'default' after 2.28 s
michaelp@MacBook-Air-5 icj-project-rig % 
...
```
- [x] No obvious formatting errors are present (unused variables, missing new lines at the end of documents, etc.)
- [x] Manually tested GitHub action: https://github.com/utdata/icj-project-rig/pull/90#discussion_r1231628496

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -15,12 +15,14 @@ import fetch from './tasks/fetch.js';
...
+    bake,
     clean,
-    styles,
+    clear,
     copy,
+    fetch,
     gulp.parallel(lint, scripts, images),
     nunjucks,
-    bake
+    styles
```
Comment (michplunkett) [inline gulpfile.js:25]: Added two missing commands and alpha-sorted them.

File: package.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
   "main": "index.html",
   "type": "module",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "format": "npx prettier --write .",
+    "lint": "npx eslint --fix ."
```
Comment (michplunkett) [inline package.json:9]: These commands can be run using the template: `npm run [scriptName]`

File: package.json [hunk 3]
```diff
@@ -14,12 +15,15 @@
   "keywords": [],
   "author": "Christian McDonald <christian.mcdonald@utexas.edu>",
   "license": "ISC",
+  "eslintIgnore": [
+    "docs/"
+  ],
```
Comment (michplunkett) [inline package.json:20]: The `docs` folder will not be linted.

File: .gitignore [hunk 4]
```diff
@@ -8,6 +8,10 @@
 .AppleDouble
 .LSOverride
 
+# IDE files
+.idea
+.vscode
```
Comment (michplunkett) [inline .gitignore:13]: Ignore standard IDE folders.

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+## Describe your changes
```
Comment (michplunkett) [inline .github/pull_request_template.md:1]: Now all PRs have a standard template.

File: tasks/nunjucks.js [hunk 6]
```diff
@@ -1,10 +1,11 @@
 import gulp from 'gulp';
+import fs from 'fs-extra';
 import nunjucksRender from 'gulp-nunjucks-render';
 import * as journalize from 'journalize';
 import browserSync from 'browser-sync';
-import config from '../project.config.json' assert {type: 'json'};
 import log from 'fancy-log';
-import fs from 'fs';
+
+const config = fs.readJson('../project.config.json');
```
Comment (michplunkett) [inline tasks/nunjucks.js:8]: https://github.com/eslint/eslint/discussions/15305#discussioncomment-2400923

File: .github/workflows/format-and-lint.yml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        with:
+          # prettier CLI arguments.
+          args: --check --config .prettierrc.json --ignore-path .prettierignore .
+  eslint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 16.18.0
```
Comment (michplunkett) [inline .github/workflows/format-and-lint.yml:24]: Use the node version that we use to compile the project.

**Comment by michplunkett** [comment]
The function has been addressed and it looks good to go. Here's a sample of the output:
```
michaelp@MacBook-Air-5 icj-project-rig % gulp
[11:24:54] Using gulpfile ~/Documents/GitHub/icj-project-rig/gulpfile.js
[11:24:54] Starting 'default'...
[11:24:54] Starting 'bake'...
[11:24:54] Finished 'bake' after 5.41 ms
[11:24:54] Starting 'clean'...
[11:24:54] Finished 'clean' after 906 μs
[11:24:54] Starting 'clear'...
[11:24:54] Finished 'clear' after 19 ms
[11:24:54] Starting 'copy'...
[11:24:54] Finished 'copy' after 3.77 ms
[11:24:54] Starting 'fetch'...
[11:24:54] Finished 'fetch' after 238 μs
[11:24:54] Starting 'lint'...
[11:24:54] Starting 'scripts'...
[11:24:54] Starting 'images'...
[11:24:55] Finished 'lint' after 31 ms
Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
[11:24:55] Finished 'scripts' after 165 ms
[11:24:55] gulp-imagemin: Minified 3 images (saved 198 kB - 65.2%)
[11:24:55] Finished 'images' after 223 ms
[11:24:55] Starting 'nunjucks'...
[11:24:55] Finished 'nunjucks' after 1.1 ms
[11:24:55] Starting 'styles'...
[11:24:57] Finished 'styles' after 1.97 s
[11:24:57] Finished 'default' after 2.23 s
...
```

File: tasks/bake.js [hunk 8]
```diff
@@ -9,7 +9,8 @@ import isValidGlob from 'is-valid-glob';
 
 // modularize manageEnv
 import * as journalize from 'journalize';
-import fs from 'fs';
+
+const config = fs.readJsonSync('./project.config.json');
```
Comment (michplunkett) [inline tasks/bake.js:13]: From what I read, Node paths are based on where you're calling the function from, so we reference the base directory here.

**Review by critmcdonald — COMMENTED** [review]
Some things:
- The `fetch` task should be explicitly user initiated and not be included in the default `gulp` task nor the `gulp dev` task. The source files could be various states of edit and the development user may not know or not want to redownload those file until they are sure they want to.

I was able to fetch and run all tasks. I'm find with you approving and merging once that change is set.

File: gulpfile.js [hunk 9]
```diff
@@ -1,15 +1,15 @@
 import gulp from 'gulp';
+import bake from './tasks/bake.js';
```
Comment (michplunkett) [inline gulpfile.js:2]: Alpha-sorted the imports.

**Comment by michplunkett** [comment]
Gulp tasks pre-change:
```
michaelp@MacBook-Air-5 icj-project-rig % gulp --tasks
(node:60209) ExperimentalWarning: Importing JSON modules is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
[14:22:01] Tasks for ~/Documents/GitHub/icj-project-rig/gulpfile.js
[14:22:01] ├── clean
[14:22:01] ├── styles
[14:22:01] ├── copy
[14:22:01] ├── lint
[14:22:01] ├── scripts
[14:22:01] ├── images
[14:22:01] ├── nunjucks
[14:22:01] ├── bake
[14:22:01] ├── serve
[14:22:01] ├── clear
[14:22:01] ├── fetch
[14:22:01] ├─┬ default
[14:22:01] │ └─┬ <series>
[14:22:01] │   ├── clean
[14:22:01] │   ├── styles
[14:22:01] │   ├── copy
[14:22:01] │   ├─┬ <parallel>
[14:22:01] │   │ ├── lint
[14:22:01] │   │ ├── scripts
[14:22:01] │   │ └── images
[14:22:01] │   ├── nunjucks
[14:22:01] │   └── bake
[14:22:01] └─┬ dev
[14:22:01]   └─┬ <series>
[14:22:01]     ├─┬ default
[14:22:01]     │ └─┬ <series>
[14:22:01]     │   ├── clean
[14:22:01]     │   ├── styles
[14:22:01]     │   ├── copy
[14:22:01]     │   ├─┬ <parallel>
[14:22:01]     │   │ ├── lint
[14:22:01]     │   │ ├── scripts
[14:22:01]     │   │ └── images
[14:22:01]     │   ├── nunjucks
[14:22:01]     │   └── bake
[14:22:01]     └── serve
michaelp@MacBook-Air-5 icj-project-rig % 
```

Gulp tasks post-change:
```
michaelp@MacBook-Air-5 icj-project-rig % gulp --tasks  
[14:20:07] Tasks for ~/Documents/GitHub/icj-project-rig/gulpfile.js
[14:20:07] ├── bake
[14:20:07] ├── clean
[14:20:07] ├── clear
[14:20:07] ├── copy
[14:20:07] ├── fetch
[14:20:07] ├── images
[14:20:07] ├── lint
[14:20:07] ├── nunjucks
[14:20:07] ├── scripts
[14:20:07] ├── serve
[14:20:07] ├── styles
[14:20:07] ├─┬ default
[14:20:07] │ └─┬ <series>
[14:20:07] │   ├── clean
[14:20:07] │   ├── styles
[14:20:07] │   ├── copy
[14:20:07] │   ├─┬ <parallel>
[14:20:07] │   │ ├── lint
[14:20:07] │   │ ├── scripts
[14:20:07] │   │ └── images
[14:20:07] │   ├── nunjucks
[14:20:07] │   └── bake
[14:20:07] └─┬ dev
[14:20:07]   └─┬ <series>
[14:20:07]     ├─┬ default
[14:20:07]     │ └─┬ <series>
[14:20:07]     │   ├── clean
[14:20:07]     │   ├── styles
[14:20:07]     │   ├── copy
[14:20:07]     │   ├─┬ <parallel>
[14:20:07]     │   │ ├── lint
[14:20:07]     │   │ ├── scripts
[14:20:07]     │   │ └── images
[14:20:07]     │   ├── nunjucks
[14:20:07]     │   └── bake
[14:20:07]     └── serve
michaelp@MacBook-Air-5 icj-project-rig % 
```

**Pull request merged**